files/justfiles/toggles.just

```diff
@@ -177,22 +177,74 @@ toggle-xwayland ACTION="prompt":
 
 # Toggle bash environment lockdown (mitigates LD_PRELOAD attacks)
 toggle-bash-environment-lockdown:
-    #!/usr/bin/bash
-    BASH_ENV_FILES=("$HOME/.bashrc" "$HOME/.bash_profile")
+    #!/bin/run0 /bin/bash
+    set -eou pipefail
+    user_string=$(ls "/var/home" | tr '\n' ',' | sed 's/,*$//')
+    IFS=',' read -ra user_list <<< "$user_string"
+    [ "$UID" -eq 0 ] || { echo "This script must be run as root."; exit 1;} #almost certainly not needed, but doesn't hurt
+    declare -a filtered_list
+    for element in "${user_list[@]}"; do
+        if [ ! -d "/var/home/$element/Desktop/" ]; then
+                break
+        fi
+        filtered_list+=("$element")
+    done
+    b=$(tput bold) #This is the more correct way to bold and satisfies -eou pipefail
+    n=$(tput sgr0)
     echo "${b}WARNING${n} This will overwrite your .bashrc and .bash_profile."
     echo "This is needed to ensure the mitigation is effective."
     echo "Do you understand?"
     echo "Please type in \"YES I UNDERSTAND\" and press enter"
-    read ACCEPT
-    if [ "$ACCEPT" == "YES I UNDERSTAND" ]; then
-      if lsattr "${BASH_ENV_FILES[0]}" 2>/dev/null | awk '{print $1}' | grep -q 'i'; then
-        echo "Bash environment '(${BASH_ENV_FILES[@]})' is locked down. Unlocking it."
-        for file in "${BASH_ENV_FILES[@]}"; do
-            run0 chattr -i "$file"
+    read -r ACCEPT
+    if [ "$ACCEPT" != "YES I UNDERSTAND" ]; then
+        exit
+    fi
+    echo "Do you want this change to apply to ${b}all${n} users? [y/N]"
+    echo "Otherwise, it will only apply to the currently logged in user."
+    read -r REPLY
+    if [[ ! "$REPLY" == [yY]* ]]; then
+        filtered_list[0]=$SUDO_USER
+        filtered_list[1]=$SUDO_USER
+        if lsattr "/etc/profile" 2>/dev/null | awk '{print $1}' | grep -q 'i'; then
+                change="unlocked"
+        else
+                change="locked"
+        fi
+    fi
+    for user in "${filtered_list[@]}"; do
+    	echo "Working user:$user"
+    	user_home=$(bash -c "cd ~$(printf %q "$user") && pwd")
+    	BASH_ENV_FILES=("/etc/profile") #used because we know it exists, not that it needs to be immutable
+    	if [ -d "$user_home/.bashrc.d/" ]; then
+            chmod -R "644" "$user_home/.bashrc.d/"
+    	fi
+    	BASH_FILES=("$user_home/.bashrc" "$user_home/.bash_profile" "$user_home/.config/bash-completion" "$user_home/.profile" "$user_home/.bash_logout" "$user_home/.bash_login" "$user_home/.config/environment.d")
+    	for file in "${BASH_FILES[@]}"; do
+            if [ -f "$file" ]; then
+                BASH_ENV_FILES+=("$file")
```

Comment:
`BASH_ENV_FILES+=("$file")` can be pulled outside of `if` since it's the same in both branches.

Reply:
Actually it's still not resolved.